tests/cond_order/tests/01-run.py

```diff
@@ -10,28 +10,31 @@
 import os
 import sys
 
-thread_prio = {
-        3:  6,
-        4:  4,
-        5:  0,
-        6:  2,
-        7:  1
-        }
-first_group_size = 3
+NR_THREADS = 5
 
 
 def testfunc(child):
 
-    for k in thread_prio.keys():
-        child.expect_exact("T{} (prio {}): waiting on condition variable now"
-                           .format(k, thread_prio[k]))
-    count = 0
-    last = -1
+    # First collect the thread info how they are created
+    # A number of lines with:
+    #  T4 (prio 6): waiting on condition variable now
+    thread_prios = {}
+    for nr in range(NR_THREADS):
+        child.expect(r"T(\d+) \(prio (\d+)\): waiting on condition variable now")
+        thr_id = int(child.match.group(1))
+        thr_prio = int(child.match.group(2))
+        thread_prios[thr_id] = thr_prio
+
     child.expect_exact("First batch was signaled")
-    for _ in range(len(thread_prio)):
-        child.expect(r"T\d+ \(prio (\d+)\): condition variable was signaled now")
-        assert(int(child.match.group(1)) > last)
-        last = int(child.match.group(1))
+
+    count = 0
+    last_prio = -1
+    for _ in range(len(thread_prios)):
+        child.expect(r"T(\d+) \(prio (\d+)\): condition variable was signaled now")
+        thr_id = int(child.match.group(1))
+        thr_prio = int(child.match.group(2))
+        assert(thr_prio > last_prio)
```

Comment:
I would at least add a check that the `thr_prio` of `thr_id` has not changed compared to the first loop, e.g.:
```suggestion
        assert(thr_prio > last_prio)
        assert(thr_prio > thread_prios[thr_id])
```
by the way, I would also give more explicit names to variables:
- `thr_prio` => `thread_prio`
- `thr_id` => `thread_id`

And for cosmetic: `NR_THREADS` => `NB_THREADS`

Reply:
OK about variable name, assert syntax, and extra check.
But `NB_THREADS`? I've never seen anyone use abbreviation `nb` for number.

Reply:
About `NB_THREADS` you are right, it seems to be more french (sorry :) ). Maybe use `NUM_THREADS` instead ?